pkg/filesystem/subvolume.go

```diff
@@ -125,12 +125,29 @@ func getK8sRefSubvolume(ctx context.Context, clientsets *k8sutil.Clientsets) map
 	subvolumeNames := make(map[string]subVolumeInfo)
 	for _, pv := range pvList.Items {
 		if pv.Spec.CSI != nil {
-			subvolumeNames[pv.Spec.CSI.VolumeAttributes["subvolumeName"]] = subVolumeInfo{}
+			subvolumePath := pv.Spec.CSI.VolumeAttributes["subvolumePath"]
+			name := getSubvolumeNameFromPath(subvolumePath)
```

Comment:
@yati1998 if we are not able to get the name from the PV we need to log it as an error so that the user knows something is wrong, silently continuing can cause problems with expectation miss match

Reply:
@Madhu-1 addressed the change.